borg/archive.py

```diff
@@ -468,6 +470,23 @@ def rename(self, name):
         del self.manifest.archives[self.name]
 
     def delete(self, stats, progress=False, forced=False):
+        class FatalError(Exception):
+            """some fatal error occurred, the transaction must be aborted."""
+
+        def chunk_decref(id, stats):
+            nonlocal error
+            try:
+                self.cache.chunk_decref(id, stats)
+            except KeyError as e:
+                # chunkid missing from chunks index - chunks cache corrupted!
+                raise FatalError(str(e))
+            except Repository.ObjectNotFound as e:
+                # object not in repo - strange, but we wanted to delete it anyway.
+                # XXX doesn't this also mean that our chunks cache is corrupted?
+                # XXX obviously it had a refcount of 1 before decref, but the chunk
+                # XXX has already been deleted from the repo.
```

Comment:
hmm, yes, guess that might happen if a repo-only check is done.
not if all archives are also checked, then lost-chunk references will get replaced with a zero chunk.
afterwards, the chunks cache would resync and not reference such lost chunks any more.